stackstate_checks_base/stackstate_checks/base/utils/http_helper.py

```diff
@@ -0,0 +1,1447 @@
+import json
+import requests_mock
+from enum import Enum
+from schematics.models import Model
+from schematics.types import StringType
+from schematics.exceptions import DataError
+from requests import Session, Request
+from requests.auth import HTTPBasicAuth
+from requests.adapters import HTTPAdapter
+from requests.packages.urllib3.util.retry import Retry
+
+try:
+    import urlparse
+except ImportError:
+    import urllib.parse as urlparse
+
+
+"""
+    Authentication Models
+"""
+
+
+class HTTPBasicAuthentication(Model):
+    username = StringType(required=True)
+    password = StringType(required=True)
+
+
+"""
+    Authentication ENUM
+"""
+
+
+class HTTPAuthenticationType(Enum):
+    BasicAuth = HTTPBasicAuthentication
+
+
+"""
+    Possible Request and Response objects
+"""
+
+
+class HTTPMethod(Enum):
+    GET = 'GET'
+    POST = 'POST'
+    PUT = 'PUT'
+    PATCH = 'PATCH'
+    DELETE = 'DELETE'
+
+
+class HTTPRequestType(Enum):
+    JSON = dict
+
+
+class HTTPResponseType(Enum):
+    PLAIN = str
+    JSON = dict
+
+
+"""
+    The HTTP Helper Common class is used for common functionality split between all the other
+    HTTP Helper classes.
+    Functionality:
+        - Mapped enums to lists
+        - Log messages
+        - Create a dictionary from a string
+"""
+
+
+class HTTPHelperCommon:
+    _verbose = False
+
+    # Mapped enum values
+    http_method_enums = [item.value for item in HTTPMethod]
```

Comment:
why not place this on HTTP method itself? You can use class variables.

Reply:
Sure, I'll move those VARIABLES

Reply:
Move to within the classes